Sources/SwiftKafka/Utilities/KafkaStatistics.swift

```diff
@@ -0,0 +1,70 @@
+import NIOConcurrencyHelpers
+import NIOCore
+
+public typealias KafkaStatistics = String
+
+public struct KafkaStatisticsSeqence: Sendable, AsyncSequence {
+    struct EmptyDelegate: NIOAsyncSequenceProducerDelegate {
```

Comment:
I was thinking whether we need anything to do with this sequence, i.e.:
1. to call `finish()` in Consumer/Producer delegate or
2. to call it here

Reply:
I think ideally what we would do here is something like we do in the `KafkaConsumer` where we enter a state in which we are still polling for events but then in this case drop any incoming statistics events so that new statistics emitted after termination of the `AsyncSequence` don't buffer in memory.

Reply:
The main problem is that there is one state machine, so one switch to `.consumptionStopped` should be enough on terminate of any sequence

Reply:
Yeah, we have to work around this here. There are two things that I can currently think of:

1. In `KafkaClient` (now `RDKafkaClient` on `main`) keep track of all the `RDKafkaEvent`s we are listening to and drop inside of `KafkaClient` when `RDKafkaEvent` is not part of the list of `RDKafkaEvent`s we are listening to -> in `KafkaConsumer` / `KafkaProducer` you would then just change what the `RDKafkaClient` should listen to
2. Have something like a `StatisticaHandler` type that has its own state and just forwards stuff to the `source` or not